src/DynamoCore/Configuration/PreferenceSettings.cs

```diff
@@ -841,8 +857,34 @@ public bool IsTrustedLocation(string path)
             {
                 string location = PathHelper.ValidateDirectory(path);
 
+                var locationDir = new DirectoryInfo(location);
                 // All subdirectories are considered trusted if the parent directory is trusted.
-                return TrustedLocations.FirstOrDefault(x => location.StartsWith(x)) != null;
+                return TrustedLocations.FirstOrDefault(x =>
+                {
+                    if (location.Equals(x))
+                    {
+                        return true;
+                    }
+
+                    if (location.StartsWith(x))
+                    {
+                        var parent = locationDir.Parent;
+                        while(parent != null) 
+                        { 
+                            if (parent.FullName.Length < x.Length)
+                            {// Path length will only decrease so no need to check anymore.
+                                return false;
+                            }
+
+                            if (parent.FullName.Equals(x))
+                            {
+                                return true;
+                            }
+                            parent = parent.Parent;
```

Comment:
so, what is the parent of a volume, will it be, null, empty string, or the same volume again?

Reply:
Parent is an object (not string)
Parent of the root of the path will be null
https://docs.microsoft.com/en-us/dotnet/api/system.io.directoryinfo.parent?view=netframework-4.8#property-value